src/cleanfigure.m

```diff
@@ -554,18 +554,28 @@ function simplifyLine(meta, handle, targetResolution)
     % The resolution is lost only beyond the multiplier magnification
     mult = 2;
 
-    % Convert data to pixel units, magnify and mark only the first
-    % point that occupies a given position
-    mask = [true; diff(round(x * xToPix * mult))~=0];
-    mask = [true; diff(round(y * yToPix * mult))~=0] | mask;
+    % Convert data to pixel units and magnify
+    dataPixel = round([x * xToPix * mult, ...
+                       y * yToPix * mult]);
 
-    % Keep end points or it might truncate whole pixels
+    % Sort the pixels
+    [dataPixelSorted, id_orig] = sortrows(dataPixel);
+
+    % Find the duplicate pixels
+    mask_sorted = [true; diff(dataPixelSorted(:,1))~=0 | ...
+                         diff(dataPixelSorted(:,2))~=0];
+
+    % Unwind the sorting
+    mask            = false(size(x));
+    mask(id_orig)   = mask_sorted;
+
+    % Set the first, last, as well as unique pixels to true
+    mask(1)         = true;
+    mask(end)       = true;
+
+    % Set NaNs to true
     inan         = isnan(x) | isnan(y);
-    df           = diff([false; inan; false]);
-    istart       = df == 1;
-    pend         = find(df == -1)-1;
-    mask(istart) = true;
-    mask(pend)   = true;
```

Comment:
I think the intuition I got about this point is that the plot coordinates do not necessarily start at exactly the beginning of the pixel. So, if you have two plot coordinates which are within one pixel distance from each other, but they start in the middle of the pixel, then you will lose one of those, while you should have retained it. 

It is basically a coordinate reference problem and I just chose the safe option of keeping the ends of a line segment. In the end, having two additional coordinates per line segment is nothing compared to the sort bug you found.